File: src/main/java/github/macro/build_info/gems/GemTag.java
package github.macro.build_info.gems;

import org.jetbrains.annotations.Nullable;

import java.util.Arrays;

/**
 * Created by Macro303 on 2019-Dec-27
 */
public enum GemTag {
	WARCRY,
	CHAOS,
	AOE,
	DURATION,
	ACTIVE,
	SUPPORT,
	COLD,
	LIGHTNING,
	FIRE,
	TRAP,
	ATTACK,
	TOTEM,
	MELEE,
	SPELL,
	AURA,
	MINION,
	CHAINING,
	PROJECTILE,
	BRAND,
	CURSE,
	TRIGGER,
	BOW,
	CHANNELLING,
	MOVEMENT,
	TRAVEL,
	BLINK,
	HERALD,
	GUARD,
	MINE,
	GOLEM,
	VAAL,
	PHYSICAL,
	STRIKE,
	NOVA;

	@Nullable
	public static GemTag value(String name) {
		return Arrays.stream(values()).filter(tag -> tag.name().replaceAll("_", " ").equalsIgnoreCase(name)).findFirst().orElse(null);
	}
}

File: src/main/kotlin/github/macro/build_info/gems/GemTag.kt
package github.macro.build_info.gems

/**
 * Created by Macro303 on 2020-Jan-13.
 */
enum class GemTag {
    WARCRY,
    CHAOS,
    AOE,
    DURATION,
    ACTIVE,
    SUPPORT,
    COLD,
    LIGHTNING,
    FIRE,
    TRAP,
    ATTACK,
    TOTEM,
    MELEE,
    SPELL,
    AURA,
    MINION,
    CHAINING,
    PROJECTILE,
    BRAND,
    CURSE,
    TRIGGER,
    BOW,
    CHANNELLING,
    MOVEMENT,
    TRAVEL,
    BLINK,
    HERALD,
    GUARD,
    MINE,
    GOLEM,
    VAAL,
    PHYSICAL,
    STRIKE,
    NOVA;

    companion object {
        fun value(name: String): GemTag? = values().firstOrNull { it.name.equals(name, ignoreCase = true) }
    }
}